File: leetDaily/3069DistributeElementsIntoTwoArrI.cpp
class Solution
{
public:
    vector<int> resultArray(vector<int> &nums)
    {
        vector<int> arr1, arr2;
        int n = nums.size();

        vector<int> ans;

        arr1.push_back(nums[0]);
        arr2.push_back(nums[1]);

        for (int i = 2; i < n; i++)
        {
            int b1 = arr1.back();
            int b2 = arr2.back();

            if (b1 > b2)
            {
                arr1.push_back(nums[i]);
            }
            else
            {
                arr2.push_back(nums[i]);
            }
        }

        for (auto x : arr1)
            ans.push_back(x);
        for (auto x : arr2)
            ans.push_back(x);

        return ans;
    }
};

File: leetDaily/3069DistributeElementsIntoTwoArrI.java
class Solution {
    public int[] resultArray(int[] nums) {
        ArrayList<Integer> arr1 = new ArrayList<>();
        ArrayList<Integer> arr2 = new ArrayList<>();
        arr1.add(nums[0]);
        arr2.add(nums[1]);
        int n = nums.length;

        for (int i = 2; i < n; i++) {
            int backA1 = arr1.getLast();
            int backA2 = arr2.getLast();
            if (backA1 > backA2) {
                arr1.add(nums[i]);
            } else {
                arr2.add(nums[i]);
            }
        }

        int[] ans = new int[n];

        int idx = 0;

        for (int x : arr1) {
            ans[idx] = x;
            idx++;
        }

        for (int x : arr2) {
            ans[idx] = x;
            idx++;
        }

        return ans;

    }
}